Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/contactdiary/ui/ContactDiaryTestFragmentViewModel.kt

```diff
@@ -0,0 +1,114 @@
+package de.rki.coronawarnapp.test.contactdiary.ui
+
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPersonEncounter
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryPerson
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryPersonEncounter
+import de.rki.coronawarnapp.contactdiary.retention.ContactDiaryDataRetentionCalculation
+import de.rki.coronawarnapp.contactdiary.storage.repo.DefaultContactDiaryRepository
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import org.joda.time.LocalDate
+import kotlin.random.Random
+
+class ContactDiaryTestFragmentViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val repository: DefaultContactDiaryRepository,
+    private val timeStamper: TimeStamper,
+    private val retentionCalculation: ContactDiaryDataRetentionCalculation
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val locationVisits = repository.locationVisits.asLiveData(context = dispatcherProvider.Default)
+    val personEncounters = repository.personEncounters.asLiveData(context = dispatcherProvider.Default)
+
+    fun getFancyLocationVisitString(list: List<ContactDiaryLocationVisit>): String {
```

Comment:
Could be an extension fun
SortedBy { it.date } is easier in this case
StringBuilder for better performance
Could use List.map and List.foreach